APCAppCore/APCAppCoreTests/Reminder Tests/APCTasksReminderManagerTests.m

```diff
@@ -297,4 +303,211 @@ - (void) testMPowerDaily
     XCTAssertEqual(NSCalendarUnitDay, saturdayAt5Pm.repeatInterval);
 }
 
+// pragma mark - old to new tasks
+
+- (void) testSurveyNoNew
+{
+    NSArray* oldTaskGroups = @[[self taskGroupWithID:@"qow"
+                                                type:APCTaskTypeSurveyTask
+                                           updatedAt:[NSDate date]
+                                            complete:NO],
+                               
+                               [self taskGroupWithID:@"activity"
+                                                type:APCTaskTypeActivityTask
+                                           updatedAt:[NSDate date]
+                                            complete:NO]];
+    
+    NSArray* newTaskGroups = @[[self taskGroupWithID:@"qow"
+                                                type:APCTaskTypeSurveyTask
+                                           updatedAt:[[NSDate date] dateByAddingDays:1]
+                                            complete:NO],
+                               
+                               [self taskGroupWithID:@"activity"
+                                                type:APCTaskTypeActivityTask
+                                           updatedAt:[[NSDate date] dateByAddingDays:1]
+                                            complete:NO]];
+    
+    NSArray* newReminderTasks = [APCTasksReminderManager findNewTasksFromOld:oldTaskGroups toNewTaskGroups:newTaskGroups];
+    
+    XCTAssertTrue(newReminderTasks != nil);
+    XCTAssertTrue(newReminderTasks.count == 0);
+}
+
+- (void) testSurveyAdded
+{
+    NSArray* oldTaskGroups = @[[self taskGroupWithID:@"activity"
+                                                type:APCTaskTypeActivityTask
+                                           updatedAt:[NSDate date]
+                                            complete:YES]];
+    
+    NSArray* newTaskGroups = @[[self taskGroupWithID:@"qow"
+                                                type:APCTaskTypeSurveyTask
+                                           updatedAt:[NSDate date]
+                                            complete:NO],
+                               
+                               [self taskGroupWithID:@"activity"
+                                                type:APCTaskTypeActivityTask
+                                           updatedAt:[NSDate date]
+                                            complete:YES]];
+    
+    NSArray* newReminderTasks = [APCTasksReminderManager findNewTasksFromOld:oldTaskGroups toNewTaskGroups:newTaskGroups];
+    
+    XCTAssertTrue(newReminderTasks != nil);
+    XCTAssertTrue(newReminderTasks.count > 0);
+    XCTAssertEqual(((APCTask*)newReminderTasks[0]).taskID, @"qow");
+}
+
+- (void) testSurveyRecurringAdded
+{
+    NSArray* oldTaskGroups = @[[self taskGroupWithID:@"qow"
```

Comment:
taskID. the "qow" gives the impression (to me) that its FPHS specific. If you are just picking any taskID, should probably be a more generic one.